### DISCUSSION TITLE
Website improvement

### DISCUSSION THREAD

**Discussion by ihhub** [ihhub/fheroes2#3738] [General]
Let's discuss our deployment website. How we can improve it and make it more user friendly.

I express my vision of the wesbite:
- the first page should contain a brief explanation of the project and the main differences (enhancements) between original HoMM2 and our product such as game resolution, instant battles, monster descriptions and etc. Right now we don't have much but in the future we will. Also Copyrights and Dontation. In other words, nothing related to compilation. We could also add explicit information about social networks as some people don't know what these small icons for :)
- one page should be dedicated to Downloads is what we have right now. The missing page is a link to the main page should be added
- one page for compilation and contribution
- one page for the list of improvements. @Branikolog is working on the list

We could keep all these pages in a separate directory from the main README.md GitHub file:
![image](https://user-images.githubusercontent.com/19829520/122791152-92625f00-d2eb-11eb-9edd-1b7c15e1e0f0.png)

This is just a draft and I would like to hear opinions of others: @Branikolog , @oleg-derevenetz , @idshibanov , @LeHerosInconnu.

**Comment by Branikolog** [comment]
@ihhub 
My vision of website 99% coincides with yours.) 
First page should describe our project (with attached screenshots), it's goals, differences with the OG and, probably the origins. Our contacts (Social networks and email) should be included as well in a separate sentence with links, as labels are not well perceptible. Here should also be mentioned the list of our project improvements comparing to the original game, I'm working at currently.

Also, I'd like to watch the "Requirements" on a main page after the description of the project. People should clearly understand before downloading, that they need the original version of the game.

After that I think we should make a link to "Download and Installation guide". Inside the link we should have two separate sections for download+install and compiling. Each section should be on a separate page. I love to way @oleg-derevenetz made a list of precompiled binaries at installation section currently. Can we make the same list for downloads of compiled versions for all OSes? 
I'm not sure we should type there a link to our pre-released versions, as common people should better download stable release version, while contributors can easily find proper links in "releases" section on the right side of the main page by themselves.
Also, download section should have a small explanation on differences of SDL1 and SDL2 versions, because this is the most commonly asked question from people within the first acquaintance with our project. Also, here we can add a tip, where archive should be unpacked to and the list, which folders of the OG are required to be in a game folder.
Another section (compiling) should be in a separate page, where we can also add a link to a "Contribution" on a main page.

"Contribution", "License" and "Donation" could be in a bottom of the main page.
Donation should also have a few sentences explaining our goals of the future development of the project, and also here we can add a link to our "Beyond 1.0" ideas, which teases some future improvements. (I Hope, @ihhub will add a few new screenshots there... ;) )

**Reply by LeHerosInconnu** [reply]
Hello everyone,

This all seems appropriate.
I would just like to suggest that the download homepage should be simple.

A text that explains that the original game or game demo files are needed.
Big buttons for the main operating systems being used that start the download directly, another button for more experienced users with more choices.

**Reply by oleg-derevenetz** [reply]
Hi @Branikolog

> I love to way @oleg-derevenetz made a list of precompiled binaries at installation section currently. Can we make the same list for downloads of compiled versions for all OSes?

I think yes, because files are not currently versioned (don't contain a version number in their names). I also think about reworking this page a little, for the purpose of moving download links for particular OS to the section of this OS, and to remove the first Download section entirely.

> I'm not sure we should type there a link to our pre-released versions, as common people should better download stable release version, while contributors can easily find proper links in "releases" section on the right side of the main page by themselves.

They are able to easily find pre-release versions only when they know what to look for :) Personally, I only found out about the existence of these pre-release versions by accident.

> Also, download section should have a small explanation on differences of SDL1 and SDL2 versions, because this is the most commonly asked question from people within the first acquaintance with our project.

I'd just always suggest installation of SDL2 version, and recommend installation of SDL1 version only if there are some issues with SDL2.

> Also, here we can add a tip, where archive should be unpacked to and the list, which folders of the OG are required to be in a game folder.

Do you have a comprehensive list of such folders? I know that there are "more complete" editions of HoMM2 and "less complete" editions. It would be nice to have a complete list of folders indicating which ones may not be available depending on the version (or edition) of the original game.

**Reply by Branikolog** [reply]
Hi, @oleg-derevenetz 

> They are able to easily find pre-release versions only when they know what to look for :) Personally, I only found out about the existence of these pre-release versions by accident.

To prevent numerous questions like: "What is the difference?" etc I propose to show only release versions after visitor clicks "Download" link on the main page. For other "curious" people we can make a small link in a bottom line of "Download" section.

> I'd just always suggest installation of SDL2 version, and recommend installation of SDL1 version only if there are some issues with SDL2.

So, we need to type this in the proper section. Something like "SDL2 is for Up-to-date OSes and SDL1 is for old OSes" or "we recommend to download SDL2 version rather than SDL1 if your OS supports it." Or even both this statements.

> Do you have a comprehensive list of such folders? I know that there are "more complete" editions of HoMM2 and "less complete" editions. It would be nice to have a complete list of folders indicating which ones may not be available depending on the version (or edition) of the original game.

I don't have such list either. But we can at least point that archive should be unpacked into a root folder of HoMM2. Probably, we can attach a small screenshot of typical H2 game folder...

**Reply by oleg-derevenetz** [reply]
@Branikolog

> I don't have such list either. But we can at least point that archive should be unpacked into a root folder of HoMM2. Probably, we can attach a small screenshot of typical H2 game folder...

There is already an instruction for those who purchased the original game "to copy all files belonging to the original game to the fheroes2 installation directory", which is effectively the same.

**Reply by Branikolog** [reply]
@oleg-derevenetz 
Oh, nice! Didn't noticed that.
I think, that's because these instructions go after install section with precompiled binaries. Could we move these instructions exactly to "Download" section? (We can rename it into "download and install" and create a personal instructions at a separate page for every OS) I predict people after downloading the proper version of fheroes2 engine and start acting on their own, without reading further instructions...

**Reply by oleg-derevenetz** [reply]
It would be nice to get at least some outline of the front page (text file with all sentences and may be basic formatting).

**Reply by Branikolog** [reply]
@oleg-derevenetz 
Nice work!

> It would be nice to get at least some outline of the front page (text file with all sentences and may be basic formatting).

I hope, @ihhub would help with definition and description of our project here.

After checking the page you've created, I noticed, that there's still "Compilation" section on a main page. Could you, please, move it to a separate page, like you did with "downloading" section? I'm not sure we should frighten people with such complicated things on the main page. Those, who are interested in compilation will definitely follow a link and read proper information there.

Also, I suggest to highlight the phrase "Bundled script" to catch an attention of visitors, that they don't need to search demo somewhere in google and everything is provided inside our package. What do you think?
![image](https://user-images.githubusercontent.com/55348946/123168627-e6eb1300-d480-11eb-8d88-c2d3c405171e.png)

One my personal opinion:
SDL2 and SDL1 version are not easily distiguished here. I'm not sure if this is the best solution, but I'd like to watch both these versions on a separate lines. Something like this:
`> Windows x64 (64-bit): SDL2 (recommended)`
`or`
`Windows x64 (64-bit): SDL1`

`> Windows x86 (32-bit): SDL2 (recommended)`
`or`
`Windows x86 (32-bit): SDL1`
![image](https://user-images.githubusercontent.com/55348946/123169660-40a00d00-d482-11eb-8a40-61c97dff6d9c.png)

And the same for Linux section: a link cut into two lines looks not good. Would you mind placing the link on a single line?
![image](https://user-images.githubusercontent.com/55348946/123170269-fff4c380-d482-11eb-90c9-7bb72ebab169.png)

**Reply by oleg-derevenetz** [reply]
Hi @ihhub and @Branikolog I recently made a few changes, please take a look:

https://oleg-derevenetz.github.io/fheroes2/

https://github.com/oleg-derevenetz/fheroes2/tree/install-md-improvement
